src/Orleans.Hosting.ServiceFabric/OrleansCommunicationListener.cs

```diff
@@ -0,0 +1,117 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Microsoft.ServiceFabric.Services.Communication.Runtime;
+using Newtonsoft.Json;
+using Orleans.Runtime;
+using Orleans.Runtime.Configuration;
+using Orleans.ServiceFabric;
+
+namespace Orleans.Hosting.ServiceFabric
+{
+    /// <summary>
+    /// Service Fabric communication listener which hosts an Orleans silo.
+    /// </summary>
+    public class OrleansCommunicationListener : ICommunicationListener
+    {
+        private readonly Action<ISiloHostBuilder> configure;
+        
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OrleansCommunicationListener" /> class.
+        /// </summary>
+        public OrleansCommunicationListener(Action<ISiloHostBuilder> configure)
+        {
+            this.configure = configure ?? throw new ArgumentNullException(nameof(configure));
+        }
+
+        /// <summary>
+        /// Gets or sets the underlying <see cref="ISiloHost"/>.
+        /// </summary>
+        /// <remarks>Only valid after <see cref="OpenAsync"/> has been invoked. Exposed for testability.</remarks>
+        public ISiloHost SiloHost { get; private set; }
```

Comment:
I would just call this property Host, so that we can hopefully then switch the type to use the generic IHost